### PR TITLE
Feat: let having support alias

### PR DESCRIPTION
**Pull request by CarolineZhang666**
@flycash ,please help to review this PR

### CODE REVIEW

**Comment by CarolineZhang666** [comment]
@flycash ,please help to review this PR

File: builder.go [hunk 1]
```diff
@@ -75,13 +75,22 @@ func (b *builder) buildExpr(expr Expr) error {
 	case RawExpr:
 		b.buildRawExpr(e)
 	case Column:
-		cm, ok := b.meta.fieldMap[e.name]
-		if !ok {
-			return internal.NewInvalidColumnError(e.name)
+		//检查别名  name
```
Comment (flycash) [inline builder.go:78]: do no use Chinese

File: builder.go [hunk 2]
```diff
@@ -75,13 +75,22 @@ func (b *builder) buildExpr(expr Expr) error {
...
+		//cMeta, ok := s.meta.fieldMap[field]
```
Comment (flycash) [inline builder.go:79]: remove this line

File: select_test.go [hunk 3]
```diff
@@ -107,6 +107,17 @@ func TestSelectable(t *testing.T) {
...
+			name:    "alias in having",
+			builder: New().Select(Columns("Id"), Columns("FirstName"),Avg("Age").As("avg_age")).From(&TestModel{}).GroupBy("FirstName").Having(C("avg_age").LT(20)),
+			wantSql: "SELECT `id`,`first_name`,AVG(`age`) AS `avg_age` FROM `test_model` GROUP BY `first_name` HAVING `avg_age`<?;",
+			wantArgs: []interface{}{20},
+		},
+		{
+			name:    "invalid alias in having",
+			builder: New().Select(Columns("Id"), Columns("FirstName"),Avg("Age").As("avg_age")).From(&TestModel{}).GroupBy("FirstName").Having(C("Invalid").LT(20)),
+			wantErr: internal.NewInvalidColumnError("Invalid"),
+		},
```
Comment (flycash) [inline select_test.go:120]: Write examples for `Selector.Having` and you can take `ExampleSelector_OrderBy` as reference

**Comment by CarolineZhang666** [comment]
@flycash ,please help to review this PR

**Comment by CarolineZhang666** [comment]
@flycash ,please help to review this PR

**Review by flycash — APPROVED** [review]

**Pull request merged**